### DISCUSSION TITLE
Tasmota suitable wifi radiator actuator thermostat ???

### DISCUSSION THREAD

**Discussion by c0nne1504683nf** [arendst/Tasmota#13059] [Q&A] [answered]
Hi all,

I'm looking for a Tasmota suitable wifi radiator actuator thermostat, but couldn't find any until now. At least not at the blakadder.com list.

Also, some kind of gateway solution between Tasmota and Blutetooth (BLE) radiator regulators, such as SilverCrest RT2000 would be interesting.

My goal would be to be able to read and control a commercially available, inexpensive, battery-powered radiator regulator via MQTT.

Does anyone know of a feasible solution? I'm grateful for any advice.

Best,
c0nne1504683nf

**Comment by barbudor** [comment]
If this is the kind of thing you are looking for : https://www.eq-3.com/products/homematic/detail/bluetooth-smart-radiator-thermostat.html
Some work has been do by @btsimonh but not sure this has been fully backported to Tasmota main repo.

You can join this conversation : https://github.com/arendst/Tasmota/discussions/10919
or on the Discord server, channel #ble-esp32 you can ask some question about it

**Reply by sfromis** [reply]
For radiator actuator thermostats, I'd be inclined to look in the direction of [Zigbee devices](https://zigbee.blakadder.com/hvac.html). ESP8266 is not a great match for battery-powered devices.

**Comment by btsimonh** [comment, marked as answer]
The work @barbudor refers to is BLE control of the [bluetooth Eq3 valves.](https://www.amazon.co.uk/eQ-3-Eqiva-BLUETOOTH%C2%AE-Smart-Heizk%C3%B6rperthermostat-White/dp/B01N39V0I4/ref=asc_df_B01N39V0I4/?tag=googshopuk-21&linkCode=df0&hvadid=311059984004&hvpos=&hvnetw=g&hvrand=16908968809018776220&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=1007127&hvtargid=pla-423158937115&psc=1)  The valves are about E20 each, you can control them via an ESP32 based Tasmota, and come in vertcial or hor.  2 x AA last about 1 year.  I've used the ESP32-Cam modules as the most available with external antenna.  They don't have accessible temperature readings - but you can set the target temp.   BLE also makes it easy to have a $3 external temp/hum sensor in each room.
I've not added the driver to the main Tas repo yet, but it is available here:
https://github.com/btsimonh/Tasmota
Personally, I'd get one and and an ESP32 tas, and check that the firmware on the valve has not been significantly changed before buying 8....
br,
Simon

**Comment by c0nne1504683nf** [comment]
Thanks @barbudor , @sfromis , @btsimonh !

Seems that btsimonh's approach is worth a trial.

Today I ordered the EQ3 from German deliverer (https://www.voelkner.de/products/1017171/eqiva-141771E0-CC-RT-BLE-EQ-Funk-Heizkoerperthermostat-elektronisch.html - should be the same as yours). What I already have are a few BME280 breakout boards. What I don't have is the recommended ESP32 board. I have two different boards: https://eckstein-shop.de/HIMALAYAMatrix-CoreESP32EntwicklerboardmitESP32-BitWiFi2BBluetoothIoTDEVBoard and https://wiki.dfrobot.com/Beetle_ESP32_SKU_DFR0575 Unfortunately, I don't know if these might be usable as well. In case of need I have to order the ESP32-Cam board.

My biggest concern is whether I can manage compiling this particular version and flashing it to the board. The only thing I have done so far with Tasmota is OTA flashing a Gosund SP112 with the default binary. But I must and will try as soon as I have all the parts.

Best,
c0nne1504683nf

**Reply by sfromis** [reply]
Which "normal" ESP32 board you use should make no difference to BLE, unless the antenna or power situation is "unusually bad".
